docs/spec/BeakerSpec.tex

```diff
@@ -0,0 +1,994 @@
+
+\usepackage{microtype}
+\usepackage[unicode=true]{hyperref}
+\usepackage[usenames,dvipsnames]{color}
+\hypersetup{breaklinks=true,
+            bookmarks=true,
+            pdfauthor={},
+            pdftitle={},
+            colorlinks=true,
+            citecolor=blue,
+            urlcolor=blue,
+            %linkcolor=magenta,
+            pdfborder={0 0 0}}
+\urlstyle{same}  % don't use monospace font for urls
+\usepackage{color}
+\usepackage{listings}
+
+\usepackage{longtable,booktabs}
+\usepackage{graphicx,grffile}
+\makeatletter
+\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
+\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
+\makeatother
+% Scale images if necessary, so that they will not overflow the page
+% margins by default, and it is still possible to overwrite the defaults
+% using explicit options in \includegraphics[width, height, ...]{}
+\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
+\setlength{\parindent}{0pt}
+\setlength{\parskip}{6pt plus 2pt minus 1pt}
+\setlength{\emergencystretch}{3em}  % prevent overfull lines
+
+
+% Redefines (sub)paragraphs to behave more like sections
+\ifx\subsubsection\undefined\else
+\let\oldparagraph\subsubsection
+\renewcommand{\subsubsection}[1]{\oldparagraph{#1}\mbox{}}
+\fi
+\ifx\subparagraph\undefined\else
+\let\oldsubparagraph\subparagraph
+\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
+\fi
+
+{\renewcommand{\arraystretch}{1.5}
+
+\setcounter{secnumdepth}{3}
+\setcounter{tocdepth}{3}
+
+\title{Beaker Specification}\label{beaker-specification}
+
+\author{Jacob Payne \and Jake O'Shannessy \and Alexey Troitskiy}
+
+\begin{document}
+
+\maketitle
+
+\begin{abstract}\label{abstract}
+
+We describe a secure and extensible operating system for smart contracts. Using
+a capability-based exokernel protocol, we can facilitate secure isolation,
+perform upgrades in a secure and robust manner, and prevent privilege escalation
+at any point in the development process. The protocol is intended to serve as an
+open standard and building block upon which more advanced and tailored security
+models can be built.
+
+\end{abstract}
+
+\newpage
+\tableofcontents
+\newpage
+
+\section{Definitions}\label{definitions}
+
+\subsection{Kernel Instance}
+A kernel instance is an Ethereum contract which using the Beaker kernel. It is
+this contract that holds the storage data etc. of the system.
+
+\section{Kernel Storage}\label{kernel-storage}
+This is a critical feature. \emph{All} data stored by the kernel is stored in a
+region of \emph{storage} called \emph{kernel storage}. \emph{No} other data can
+be stored in \emph{kernel storage}.
+
+This kernel storage is defined as all storage locations within a certain storage
+range. All values in kernel storage share the same storage key prefix. This
+prefix is currently 0x00, and so currently includes all values in the interval
```

Comment:
Let's increase to 4 bytes for the storage prefix and `0x ff ff ff ff` for the value. That way users can start with storage at `0x00` and they don't have to stumble immediately with kernel storage.

Reply:
Done.